File: AvaloniaBinding/MainWindow.axaml.fs
namespace AvaloniaBinding

open System
open Avalonia
open Avalonia.Controls
open Avalonia.Markup.Xaml

type MainWindow () as this =
    inherit Window ()

    do this.InitializeComponent()

    member private this.InitializeComponent() =
        AvaloniaXamlLoader.Load(this)

    member this.TextBlock : TextBlock = this.FindControl<TextBlock>("TextBlock")

File: CSAvaloniaBinding/MainWindow.axaml.cs
using Avalonia;
using Avalonia.Controls;
using Avalonia.Markup.Xaml;

namespace CSAvaloniaBinding
{
    public partial class MainWindow : Window
    {
        public MainWindow()
        {
            InitializeComponent();
        }

        private void InitializeComponent()
        {
#if DEBUG
            this.AttachDevTools();
#endif
            AvaloniaXamlLoader.Load(this);
        }
    }
}